pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            args = (numpy.empty_like(a), s, axes) + args
```

Comment:
`[update_arrays](https://github.com/mforbes/pyFFTW/blob/patch-1/pyfftw/pyfftw.pyx#L1446)`, which is used by `FFTW.__call__` on line 120 of `_utils.py` is very conservative and will reject erroneous arrays (i.e. arrays different to what the FFTW object planned for) with a `ValueError`.